test/tasks_screen/tasks_screen_test.dart

```diff
@@ -22,16 +25,90 @@ void main() {
     });
   }
 
+  Future<void> longPumpUntilSettle(WidgetTester tester) async {
+    await tester.pump();
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+  }
+
+  Future<void> populateTaskWidgets(WidgetTester tester) async {
+    await tester.runAsync(() async {
+      for (Task task in mockTaskList) {
+        await addTask(task);
+      }
+      await tester.idle();
+      await tester.pump();
+    });
+  }
+
   group('TasksScreen', () {
     setUp(() {
       setFirestoreInstance(MockFirestoreInstance());
-      addTask(mockTask);
     });
 
     testWidgets('works on all screen sizes', (WidgetTester tester) async {
       for (Size size in allDeviceSizes) {
         await initTasksScreen(tester, size);
       }
     });
+
+    testWidgets('displays all TaskWidgets', (WidgetTester tester) async {
+      await initTasksScreen(tester);
+      await populateTaskWidgets(tester);
+
+      // Need to add "skipOffstage: false" to find off-screen TaskWidgets
+      expect(find.byType(TaskWidget, skipOffstage: false), findsNWidgets(10));
+    });
+
+    group('overflow menu', () {
+      testWidgets('displays menu items when tapped', (WidgetTester tester) async {
+        await initTasksScreen(tester);
+        await longPumpUntilSettle(tester);
+
+        await tester.tap(find.byType(AppBarOverflowMenu));
+        await longPumpUntilSettle(tester);
+
+        expect(find.text('Delete All Tasks'), findsOneWidget);
+      });
+
+      testWidgets('tapping "Delete All Tasks" will remove all TaskWidgets', (WidgetTester tester) async {
+        await initTasksScreen(tester);
+        await populateTaskWidgets(tester);
+
+        await tester.runAsync(() async {
+          await tester.tap(find.byType(AppBarOverflowMenu));
+          await longPumpUntilSettle(tester);
+          await tester.tap(find.text('Delete All Tasks'));
+          await longPumpUntilSettle(tester);
+          await longPumpUntilSettle(tester);
+        });
+
+        // Currently not working, but this seems to be due to a bug in flutter_test
+        // https://github.com/flutter/flutter/issues/78832
+        await initTasksScreen(tester);
+
+        expect(find.byType(TaskWidget), findsNothing);
+
+        // Tasks get deleted from DB, but does not seem to get updated on page
+      });
+
+      testWidgets('collapses after tapping off', (WidgetTester tester) async {
+        await initTasksScreen(tester);
+
+        await tester.runAsync(() async {
+          expect(find.text('Delete All Tasks'), findsNothing);
+
+          await tester.tap(find.byType(AppBarOverflowMenu));
+          await longPumpUntilSettle(tester);
+
+          expect(find.text('Delete All Tasks'), findsOneWidget);
+
+          await tester.tapAt(Offset.zero);
+          await longPumpUntilSettle(tester);
```

Comment:
Extract this to `tapOff` method.  Everything should be a human-readable method in e2e

Reply:
Done, this isn't e2e though lol